### PR TITLE
Clean up python version and add test structure

### PR DESCRIPTION
**Pull request by willingc**

### CODE REVIEW

**Comment by rgbkrk** [comment]
I'm assuming with this we'll do unit tests on Travis and our integration tests on Circle CI?

**Comment by rgbkrk** [comment]
Thanks for cleaning up my initial experiments into using mypy by the way. 😄

**Comment by willingc** [comment]
@rgbkrk Ultimately was able to leave in versioneer by adjusting how travis sets up the environment for tests in python.

**Comment by rgbkrk** [comment]
Wahoo! Thank you @willingc!

**Comment by rgbkrk** [comment]
I'll do a review here and get this merged in soon so I can rebase the other PR.

File: bookstore/_version.py [hunk 1]
```diff
@@ -51,8 +51,8 @@ class NotThisMethod(Exception):
     """Exception raised if a method is not valid for the current scenario."""
 
 
-LONG_VERSION_PY: dict = {}
-HANDLERS: dict = {}
+LONG_VERSION_PY = {}
+HANDLERS = {}
```
Comment (rgbkrk) [inline bookstore/_version.py]: I like that you added this file to the exceptions list and then removed my half-hearted attempt to semi type this.

File: setup.py [hunk 2]
```diff
@@ -13,7 +15,7 @@
 https://github.com/pypa/sampleproject
 
 """
-from __future__ import print_function
+# from __future__ import print_function
```
Comment (rgbkrk) [inline setup.py]: We can delete the extra comments here.

File: setup.py [hunk 3]
```diff
@@ -3,6 +3,8 @@
 """"
 setup.py
 
+bookstore may be installed on Python 3.6 or higher.
+
 Note: Do a version check for IPython.
     IPython v6+ no longer supports Python 2.
     If Python 2, install ipython 5.x.
```
Comment (rgbkrk) [inline setup.py]: We can delete this note too now. 😄

**Comment by rgbkrk** [comment]
Thanks for all the structure for testing here. 😃

**Pull request merged**